### DISCUSSION TITLE
Using lobsterPyGUI on UBUNTU

### DISCUSSION THREAD

**Discussion by irfanulas** [QuantumChemist/LobsterPyGUI#3] [General]
Hello,

LobsterPyGUI does not work on UBUNTU, even though I followed the instructions given!

**Comment by QuantumChemist** [comment]
Hi @irfanulas 😄 ,

this problem might be more specific to the Ubuntu distribution version you are using. I compiled the binary on a Ubuntu 22.04.4 LTS system and can run it without any problem on my own laptop with Ubuntu 22.04.3 LTS.

Also when you download the binary, you have to change the file permissions to make it executable e.g. via `chmod u+x lobsterpygui` in the terminal (in the directory where you put the downloaded binary).

If you could give me more specific details (e.g. problems with GLIBC or GLIBCXX) about your issue, I could also help you with compiling your own binary or find another solution.

**Reply by irfanulas** [reply]
Hi @QuantumChemist,

Sorry for the delay!

The Ubuntu distribution I am using is Ubuntu 20.04.6 LTS with GLIBC 2.31-0ubuntu9.16 and  libQt5core.so.5, libQt5core.so.5.12, libQt5core.so.5.12.8.

When I execute the binary ( ./lobsterpygui), I get below warning:

./lobsterpygui: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by ./lobsterpygui)
./lobsterpygui: /lib/x86_64-linux-gnu/libQt5Core.so.5: version `Qt_5.15' not found (required by ./lobsterpygui)

I believe upgrading the GLIBC version is not good idea. Therefore, I will be happy if you help me either to compile my own binary or find another solution as you mentioned.

Thanks a lot for your patience and valuable time in advance.

**Reply by QuantumChemist** [reply]
Hi @irfanulas,

Actually, one cannot upgrade the `GLIBC` version that easily because that needs an update of the whole Linux distribution to 22.04.

But now I have compiled an Ubuntu 20.04.6 LTS version that you can find [here](https://github.com/QuantumChemist/LobsterPyGUI/releases/download/Ubuntu20.04.6LTS/lobsterpygui).

Do not forget to change the file permissions to make it executable e.g. via `chmod u+x lobsterpygui`! 😄 

And regarding Qt_5.15, you can install this by:
```bash
sudo apt install qt5-default
sudo apt install libqt5widgets5 libqt5gui5 libqt5core5a
sudo apt install qtbase5-dev qtbase5-dev-tools

```

I hope that this version works for you!

Let me know if you need more help 😄

**Reply by irfanulas** [reply]
Thanks a lot dear Christina for your help, very appreciated. I will contact you if I have any question regarding lobsterPy.

Best,
Irfan

**Reply by QuantumChemist** [reply]
Dear Irfan,

I'm glad I could help ☺️
I will close this discussion then and in case you have more questions, you're welcome to open a new discussion or an [issue](https://github.com/QuantumChemist/LobsterPyGUI/issues)! 
And don't forget that information regarding LobsterPy is found  [here](https://github.com/JaGeo/LobsterPy) 😄

**Reply by irfanulas** [reply]
Hi Christina,
It seems the lobsterpygui is not functioning. I downloaded it from the link you provided and  then made it executable. I open it with ./lobsterpugui command and it is working so far. After opening lobsterpygui window I navigate it to my lobster output files by clicking 'Open ..'  botton on the lobsterpygui window. Then try to execute one of the plotting options, e.g., 'Plot LOBSTER DOS'. But it does not functioning.  The lobster output files generated by using losbster5.1.0 version and the DFT code that  I am using is QuantumEspresso/7.2.

Just want to make sure if I miss anything or doing something wrong. 

Thanks,
Irfan

**Reply by irfanulas** [reply]
Hello, Christina,
I used lobster5.0.0 to generate data from my DFT calculation with quantumespresso/7.2. And for the analysis I used lobsterpygui compiled for Ubuntu 20.04.6 LTS. ./lobsterpygui does not work even though it opens 'gui-windows'.  I can't figure out where the problem is, any chance to check the Ubuntu 20.04.6 LTS version you sent?

I have attached a picture of the opened 'gui-window'to this post!

Thank you for your patience.

![lobsterpygui](https://github.com/user-attachments/assets/63ec3dec-30b6-4bea-be85-6af9dc2f105d)

**Reply by QuantumChemist** [reply]
Dear Irfan,
how you clicked on "Open" to navigate to the folder with your files? And then tried something else?
[Screencast from 16.09.2024 14:35:06.webm](https://github.com/user-attachments/assets/24451d2d-be25-4e5c-8c0a-41cbc3fd1206)
